gossip/dummy_tx_pool.go

```diff
@@ -99,3 +143,27 @@ func (p *dummyTxPool) SampleHashes(max int) []common.Hash {
 	}
 	return res
 }
+
+func (p *dummyTxPool) Count() int {
+	return len(p.pool)
+}
+
+func (p *dummyTxPool) Clear() {
+	p.lock.Lock()
+	defer p.lock.Unlock()
+	if len(p.pool) != 0 {
+		p.pool = p.pool[:0]
+	}
+}
+
+func (p *dummyTxPool) Delete(needle common.Hash) {
+	p.lock.Lock()
+	defer p.lock.Unlock()
+	notErased := make([]*types.Transaction, 0, len(p.pool)-1)
+	for _, tx := range p.pool {
```

Comment:
can we set `notErased[i] = tx`  instead of `notErased = append(notErased, tx)` as capacity of `notErased` is known?
```
notErased := make([]*types.Transaction, 0, len(p.pool)-1)
for i, tx := range p.pool {
	if tx.Hash() != needle {
		notErased[i] = tx
	}
}
```